### PR TITLE
1.1.x branch - Add SSL functionality

### PR DESCRIPTION
**Pull request by cfriaszapater**
Add functionality to connect to cassandra with SSL enabled (encryption and mutual authentication).

### CODE REVIEW

**Comment by cfriaszapater** [comment]
Could someone provide help with the failed checks? I don't know how to fix them, eg:
[ERROR] Failed to execute goal org.apache.rat:apache-rat-plugin:0.12:check (default) on project storm: Too many files with unapproved license: 1 See RAT report in: /home/travis/build/apache/storm/target/rat.txt -> [Help 1]

**Comment by danny0405** [comment]
@cfriaszapater 
You can see the clojure test cases reports and logs in module targets dir, also the check style plugin logs.

You can see the Java Junit logs in the console.

File: external/storm-cassandra/README.md [hunk 1]
```diff
@@ -24,6 +24,12 @@ The following properties may be passed to storm configuration.
 | **cassandra.retryPolicy**                    | -               | DefaultRetryPolicy  |
 | **cassandra.reconnectionPolicy.baseDelayMs** | -               | 100 (ms)            |
 | **cassandra.reconnectionPolicy.maxDelayMs**  | -               | 60000 (ms)          |
+| **cassandra.ssl.security.protocol**          | -               |                     |
+| **cassandra.ssl.keystore.path**              | -               |                     |
+| **cassandra.ssl.keystore.password**          | -               |                     |
+| **cassandra.ssl.truststore.path**            | -               |                     |
+| **cassandra.ssl.truststore.password**        | -               |                     |
+
```
Comment (danny0405) [inline external/storm-cassandra/README.md]: Can we have detailed msgs here?

File: external/storm-cassandra/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+/.checkstyle
```
Comment (danny0405) [inline external/storm-cassandra/.gitignore]: Why to add checkstyle here?

**Comment by srdo** [comment]
The RAT plugin failure is because the SslProps file doesn't have the Apache license at the top

File: external/storm-cassandra/src/main/java/org/apache/storm/cassandra/client/ClusterFactory.java [hunk 3]
```diff
@@ -67,7 +83,57 @@ protected Cluster make(Map<String, Object> stormConf) {
...
+     */
+    private static Builder configureIfSsl(Builder clusterBuilder, SslProps sslProps) {
+        if (sslProps == null || !sslProps.isSsl()) {
+            return clusterBuilder;
+        }
+
+        SSLContext sslContext = getSslContext(sslProps.getTruststorePath(), sslProps.getTruststorePassword(), sslProps.getKeystorePath(),
+                sslProps.getKeystorePassword());
+
+        // Default cipher suites supported by C*
```
Comment (erikdw) [inline external/storm-cassandra/src/main/java/org/apache/storm/cassandra/client/ClusterFactory.java]: Any purpose for this comment being retained?

Comment (cfriaszapater) [reply external/storm-cassandra/src/main/java/org/apache/storm/cassandra/client/ClusterFactory.java]: no need for it, I'll remove it

**Comment by srdo** [comment]
@cfriaszapater The failure is because the integration test POM is pointing at Storm 1.2.0-SNAPSHOT, but this branch has Storm at version 1.1.3-SNAPSHOT. You should be able to just replace the version string here https://github.com/apache/storm/blob/1.1.x-branch/integration-test/pom.xml#L25.

**Comment by cfriaszapater** [comment]
Ah, that's strange, because I did not change anything out of external/storm-cassandra. So that should be failing before this pull request. Anyway, I'll try what you say, thanks.

**Comment by HeartSaVioR** [comment]
@cfriaszapater 
Thanks for the contribution. Basically we receive PRs which go first to the master branch, and additional PRs (against 1.x-branch) which is expected to not cleanly ported back. 
Could you craft a pull request against master branch?

**Comment by cfriaszapater** [comment]
Hi HeartSaVioR,

I saw your comment and understand the reason. Just that I haven't had time to do it yet, as I had problems building the full storm project master branch locally. I'll try to do that pull request in the future to master branch.

I did this to 1.1.x branch because we use it in our organization.

Regards.

File: external/storm-cassandra/README.md [hunk 4]
```diff
@@ -24,6 +24,12 @@ The following properties may be passed to storm configuration.
 | **cassandra.retryPolicy**                    | -               | DefaultRetryPolicy  |
 | **cassandra.reconnectionPolicy.baseDelayMs** | -               | 100 (ms)            |
 | **cassandra.reconnectionPolicy.maxDelayMs**  | -               | 60000 (ms)          |
+| **cassandra.ssl.security.protocol**          | eg: null, SSL, SASL_SSL |                     |
```
Comment (danny0405) [inline external/storm-cassandra/README.md]: Can we let  the `| ` in the final line vertically alignment ?

File: external/storm-cassandra/README.md [hunk 5]
```diff
@@ -24,6 +24,12 @@ The following properties may be passed to storm configuration.
...
+| **cassandra.ssl.keystore.path**              | path to keystore |                     |
+| **cassandra.ssl.keystore.password**          | keystore password |                     |
+| **cassandra.ssl.truststore.path**            | path to truststore |                     |
+| **cassandra.ssl.truststore.password**        | truststore password |                     |
+
```
Comment (danny0405) [inline external/storm-cassandra/README.md]: Same as above

File: external/storm-cassandra/src/main/java/org/apache/storm/cassandra/client/SslProps.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.storm.cassandra.client;
+
+/**
+ * Properties needed for enabling SSL connection to Cassandra.<br/>
+ * 
+ * @author c.friaszapater
+ *
```
Comment (danny0405) [inline external/storm-cassandra/src/main/java/org/apache/storm/cassandra/client/SslProps.java]: Should remove the author name is the doc.

File: integration-test/pom.xml [hunk 7]
```diff
@@ -22,7 +22,7 @@
     <parent>
         <artifactId>storm</artifactId>
         <groupId>org.apache.storm</groupId>
-        <version>1.2.0-SNAPSHOT</version>
+        <version>1.1.3-SNAPSHOT</version>
```
Comment (danny0405) [inline integration-test/pom.xml]: Why just make the version lower?

**Comment by danny0405** [comment]
@cfriaszapater  
You should first build and install your dependencies modules successfully.

**Comment by cfriaszapater** [comment]
Hi danny0405,

In response to HeartSaVioR, I said:
> I'll try to do that pull request in the future to master branch.
> I did this to 1.1.x branch because we use it in our organization.

So, I understand I should close this pull request to 1.1.x branch, and do one to master branch.

Anyway, we use 1.1 branch version of storm-cassandra with the changes, and they work fine. I don't know why other storm modules build fails. I will try on master branch if I have time...

Regards.

**Comment by danny0405** [comment]
@cfriaszapater 
Ok, look forward to your master branch patch.

**Pull request closed without merging**